internal_transfers/actions/src/parse.ts

```diff
@@ -23,32 +18,44 @@ export function partitionEventLogs(logs: Log[]): ClassifiedEvents {
     trades: [],
     settlements: [],
   };
+  const settlementInterface = new ethers.Interface(SETTLEMENT_ABI);
+  const erc20Interface = new ethers.Interface(IERC20_ABI);
   logs.forEach((log, index) => {
-    const topics = log.topics;
-    const eventTopic = topics[0];
-
-    if (eventTopic === TRANSFER_EVENT_TOPIC) {
+    // We are only interested in Transfer Events from erc20 contracts
+    // along with Settlement and Trade Events from the Settlement contract
+    // All other event emissions can be ignored for our purposes.
+    const transferEventLog = erc20Interface.parseLog(log);
+    const settlementEventLog = settlementInterface.parseLog(log);
+    if (transferEventLog != null && transferEventLog.name === "Transfer") {
```

Comment:
`transferEventLog != null`

This wasn't here before. Why can this be null?

Reply:
In the two lines above we are attempting to parse the one event into transfer and also settlement. This event is either one of the two or neither. When it's not a transfer it will be null.